Userland/Libraries/LibJS/AST.cpp

```diff
@@ -2186,71 +2373,115 @@ Completion AssignmentExpression::execute(Interpreter& interpreter, GlobalObject&
     if (m_op == AssignmentOp::Assignment) {
         // AssignmentExpression : LeftHandSideExpression = AssignmentExpression
         return m_lhs.visit(
+            // 1. If LeftHandSideExpression is neither an ObjectLiteral nor an ArrayLiteral, then
             [&](NonnullRefPtr<Expression>& lhs) -> ThrowCompletionOr<Value> {
+                // a. Let lref be the result of evaluating LeftHandSideExpression.
+                // b. ReturnIfAbrupt(lref).
                 auto reference = TRY(lhs->to_reference(interpreter, global_object));
 
                 Value rhs_result;
+
+                // c. If IsAnonymousFunctionDefinition(AssignmentExpression) and IsIdentifierRef of LeftHandSideExpression are both true, then
                 if (lhs->is_identifier()) {
+                    // i. Let rval be NamedEvaluation of AssignmentExpression with argument lref.[[ReferencedName]].
                     auto& identifier_name = static_cast<Identifier const&>(*lhs).string();
                     rhs_result = TRY(interpreter.vm().named_evaluation_if_anonymous_function(global_object, m_rhs, identifier_name));
-                } else {
+                }
+                // d. Else,
+                else {
+                    // i. Let rref be the result of evaluating AssignmentExpression.
+                    // ii. Let rval be ? GetValue(rref).
                     rhs_result = TRY(m_rhs->execute(interpreter, global_object)).release_value();
                 }
 
+                // e. Perform ? PutValue(lref, rval).
                 TRY(reference.put_value(global_object, rhs_result));
 
+                // f. Return rval.
                 return rhs_result;
             },
             [&](NonnullRefPtr<BindingPattern>& pattern) -> ThrowCompletionOr<Value> {
+                // 3. Let rref be the result of evaluating AssignmentExpression.
```

Comment:
Should step 2: `2. Let assignmentPattern be the AssignmentPattern that is covered by LeftHandSideExpression.` Be added just before this lambda?

Reply:
Done!